mllib/src/test/scala/org/apache/spark/ml/clustering/KMeansSuite.scala

```diff
@@ -182,6 +224,7 @@ object KMeansSuite {
     "predictionCol" -> "myPrediction",
     "k" -> 3,
     "maxIter" -> 2,
-    "tol" -> 0.01
+    "tol" -> 0.01,
+    "initialModel" -> generateRandomKMeansModel(3, 3)
```

Comment:
It would be nicer to change `testEstimatorAndModelReadWrite` to accept `estimatorTestParams` and `modelTestParams` separately so we don't have to hard code certain params to be filtered out inside that method. Though we wouldn't have to that in this PR.